mne_bids/tests/test_write.py

```diff
@@ -894,3 +894,70 @@ def test_write_anat(_bids_validate):
         write_anat(output_path, subject_id, t1w_mgh, session_id, raw=raw,
                    trans=trans, verbose=True, deface=dict(theta=100),
                    overwrite=True)
+
+    # Write some MRI data and supply `landmarks`
+    mri_voxel_landmarks = mne.channels.make_dig_montage(
+        lpa=[66.08580, 51.33362, 46.52982],
+        nasion=[41.87363, 32.24694, 74.55314],
+        rpa=[17.23812, 53.08294, 47.01789],
+        coord_frame='mri_voxel')
+
+    mri_landmarks = mne.channels.make_dig_montage(
+        lpa=[-0.07629625, -0.00062556, -0.00776012],
+        nasion=[0.00267222, 0.09362256, 0.03224791],
+        rpa=[0.07635873, -0.00258065, -0.01212903],
+        coord_frame='mri')
+
+    meg_landmarks = mne.channels.make_dig_montage(
+        lpa=[-7.13766068e-02, 0.00000000e+00, 5.12227416e-09],
+        nasion=[3.72529030e-09, 1.02605611e-01, 4.19095159e-09],
+        rpa=[7.52676800e-02, 0.00000000e+00, 5.58793545e-09],
+        coord_frame='head')
+
+    meg_landmarks.save('meg_landmarks.fif')
+
+    # test mri voxel landmarks
+    anat_dir = write_anat(output_path, subject_id, t1w_mgh, session_id, acq,
+                          deface=True, landmarks=mri_voxel_landmarks,
+                          verbose=True, overwrite=True)
+    _bids_validate(output_path)
+
+    # test mri landmarks
+    with pytest.warns(UserWarning, match='Using provided `landmarks`'):
```

Comment:
if raw is then fully ignore you should just crash as raw is not useful. Warning are often not read by users.